genie-common-internal/src/main/java/com/netflix/genie/common/internal/tracing/brave/impl/EnvVarBraveTracePropagatorImpl.java

```diff
@@ -0,0 +1,224 @@
+/*
+ *
+ *  Copyright 2021 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.common.internal.tracing.brave.impl;
+
+import brave.propagation.TraceContext;
+import com.netflix.genie.common.internal.tracing.TracePropagator;
+import com.netflix.genie.common.internal.tracing.brave.BraveTracePropagator;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Implementation of {@link TracePropagator} based on <a href="https://github.com/openzipkin/brave">Brave</a> and
+ * <a href="https://github.com/openzipkin/b3-propagation">B3 Propagation</a>. This particular implementation leverages
+ * environment variables to pass context between processes.
+ * <p>
+ * Note: This current implementation kind of breaks the contract as we don't propagate all the expected headers.
+ *
+ * @author tgianos
+ * @since 4.0.0
+ */
+public class EnvVarBraveTracePropagatorImpl implements BraveTracePropagator {
+
+    /**
+     * The key for the active trace id low set of bits.
+     *
+     * @see TraceContext.Builder#traceId(long)
+     */
+    static final String GENIE_AGENT_B3_TRACE_ID_LOW_KEY = "GENIE_AGENT_B3_TRACE_ID_LOW";
+
+    /**
+     * The key for the active trace id high set of bits.
+     *
+     * @see TraceContext.Builder#traceIdHigh(long)
+     */
+    static final String GENIE_AGENT_B3_TRACE_ID_HIGH_KEY = "GENIE_AGENT_B3_TRACE_ID_HIGH";
+
+    /**
+     * The key for the parent span.
+     *
+     * @see TraceContext.Builder#parentId(long)
+     */
+    static final String GENIE_AGENT_B3_PARENT_SPAN_ID_KEY = "GENIE_AGENT_B3_PARENT_SPAN_ID";
+
+    /**
+     * The key for whether or not spans should be sampled.
+     *
+     * @see TraceContext.Builder#sampled(Boolean)
+     */
+    static final String GENIE_AGENT_B3_SAMPLED_KEY = "GENIE_AGENT_B3_SAMPLED";
+
+    /**
+     * The key for the active trace id low set of bits.
+     *
+     * @see TraceContext.Builder#traceId(long)
+     */
+    static final String GENIE_JOB_B3_TRACE_ID_LOW_KEY = "GENIE_B3_TRACE_ID_LOW";
```

Comment:
Why call it 'trace_id_low' (instead of 'trace_id'), when B3 just calls it traceId?

Reply:
I went back and forth on this a few times (and it's certainly not set in stone). Ideally I'd have just done `GENIE_B3_TRACE_ID` and used the hex string however the builder for `TraceContext` [doesn't expose the method](https://github.com/openzipkin/brave/blob/master/brave/src/main/java/brave/propagation/TraceContext.java#L416-L417) which would allow me to reconstitute the context from that single value. Hence I had to break it up and just felt it was a little clearer to use `LOW` and `HIGH` rather than absence of `LOW` representing low. That felt to me like a historical thing where they went from supporting only 64 bit identifiers to both that and 128 bit.